plugins/callback/diy.py

```diff
@@ -0,0 +1,1421 @@
+# -*- coding: utf-8 -*-
+
+# Copyright: (c) 2019, Trevor Highfill <trevor.highfill@outlook.com>
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+DOCUMENTATION = r'''
+  callback: diy
+  callback_type: stdout
+  short_description: Customize the output
+  description:
+    - Callback plugin for the DIYer, because it allows you to supply your own custom callback templates to be output
+  version_added: "2.10"
+  author: Trevor Highfill (@theque5t)
+  extends_documentation_fragment:
+    - default_callback
+  notes:
+    - Uses the L(default,./default.html) callback plugin output when a custom callback message(C(msg)) is not provided.
+    - Makes the callback event data available via the C(ansible_callback_diy) dictionary, which can be used in the templating context for the options.
+      The dictionary is only available in the templating context for the options. It is not a variable that is available via the other
+      various execution contexts, such as playbook, play, task etc.
+    - Options being set by their respective variable input can only be set using the variable if the variable was set in a context that is available to the
+      respective callback.
+      Use the C(ansible_callback_diy) dictionary to see what is available to a callback. Additionally, C(ansible_callback_diy.top_level_var_names) will output
+      the top level variable names available to the callback.
+    - Each option value is rendered as a template before being evaluated. This allows for the dynamic usage of an option. For example,
+      C("{{ 'yellow' if ansible_callback_diy.result.is_changed else 'bright green' }}")
+    - For each respective callbacks C(msg) option, the following conditions result in the following the effects. If the option value C(is None or omit)
+      then the option is not being used, and therefore use the L(default,./default.html) callback plugin. If the option value C(is not None and not omit and
+      length is not greater than 0) then the option is being used without output, and therefore suppress output. If the option value C(is not None and not omit
+      and length is greater than 0) then the option is being used with output, and therefore output.
+    - "Valid color values: C(black), C(bright gray), C(blue), C(white), C(green), C(bright blue), C(cyan), C(bright green), C(red), C(bright cyan),
+      C(purple), C(bright red), C(yellow), C(bright purple), C(dark gray), C(bright yellow), C(magenta), C(bright magenta), C(normal)"
+  seealso:
+    - name: default – default Ansible screen output
+      description: The official documentation on the B(default) callback plugin.
+      link: ./default.html
```

Comment:
It's probably best to link to https://docs.ansible.com/ansible/latest/plugins/callback/default.html for now.